### ISSUE TITLE
INFRAELECACC initializing with incorrect value

### ISSUE THREAD

**Issue by dkbohl** [PardeeCenterDU/IFs-Issues-Tracking#47]
**Short Description:** INFRAELECACC initializing with incorrect value

**Date found:** 05/05/21

**IFs Version:** 7.65 VB6 Install

**Found by:** Taylor Hanna 

**Error Type:** Incorrect initialization 

**Description:** In a model rebased to 2014, electricity access (INFRAELECACC{Total}) for Yemen initializes at 49.4, but 2014 data is 66.1.

**Steps to Replicate:**
- Install 7.66
- Rebase to 2014
- Look at INFRAELECACC{_Total_}
- Compare with _SeriesEnElecAccess%National_

**Screenshot:** 
![image](https://user-images.githubusercontent.com/12075041/117180892-aa653a00-ada2-11eb-8399-9fb2f0f24e56.png)

![image](https://user-images.githubusercontent.com/12075041/117181552-5a3aa780-ada3-11eb-855f-8b4fdf57679b.png)

**Comment by mtirfan66** [comment]
Electricity access  data series (all three - national, urban and rural)  updated in June 2020 have much better coverage now compared to the skimpy dataset we were using during the Infrastructure volume (4) of PPHP and in the following years. Most countries now have very good data coverage between 2014 and 2018. Almost all countries have 2015 and 2014.

Fix:
Pre-processor code is revised to read the base year data-point at first, for all three electricity access variables - national, urban and rural.
Growth rate will be used to initialize these variables _only when_ base year data is missing. Growth rate computation stays the same in the revised code. Computation of access rate from growth rate also stays the same, i.e., annual  growth rate is applied to the  most recent data, as long as the data is not very (more than 5 years) old compared to the base year, compounding growth/decline for the required number of years forward/backward.

Other code, e.g., regression estimation when there is no data point stays the same.

Note: 
a. At least three countries - Afghanistan, Kenya, Uganda - are having big pushes recently. The model initializes these countries with 2015 data. The 2016 to 2018 forecasts are much lower than data. These countries will look much better with a 2017 rebasing.
b. Like Yemen, Myanmar has a larger initial value by using the base year data, as opposed to growth rate based computation. I have pasted a picture of Yemen plots with old code, revised code and revised code with all historical data. 
c. LIC forecasts looks much better in terms of continuation of history (once again, I pasted an image).
d. I will send the **code changes** to Jose. Pasting screenshot below. Code change marker: 'mti 2021/05/10
There are three places of change in datainfra.bas, 
and one change in function2.bas. 
Function2.bas change was needed because of a function call (getYearAroundOrRecent) that is used elsewhere and now introduced in datainfra. The record sets from the datainfra module could be passed to that function routine, only after the default passing mechanism (ByRef) was changed to ByVal for the first argument, a record set.



![image](https://user-images.githubusercontent.com/53452620/117722943-96507c80-b196-11eb-98ab-93d3f791decd.png)

![image](https://user-images.githubusercontent.com/53452620/117723926-e4b24b00-b197-11eb-975a-c36d312a99d1.png)

![image](https://user-images.githubusercontent.com/53452620/117724041-0ad7eb00-b198-11eb-8d5c-31d17c66dddb.png)
